assets/src/amp-story-editor-blocks.js

```diff
@@ -36,6 +36,9 @@ subscribe( () => {
 	setDefaultBlockName( getSelectedBlockClientId() ? 'amp/amp-story-text' : 'amp/amp-story-page' );
 } );
 
+// Remove all blocks that aren't whitelisted.
+getBlockTypes().filter( ( { name } ) => ! ALLOWED_BLOCKS.includes( name ) ).map( ( { name } ) => unregisterBlockType( name ) );
```

Comment:
Maybe this should also be inside the `domReady`, trying to get block types from console still receives all the blocks, maybe they are not registered at that point yet? Also, probably `amp/amp-story-page` should also be excluded -- it doesn't seem to be included in the ALLOWED_BLOCKS at this moment.

Reply:
This is very well possible. Unfortunately this leads to some errors in the console.

Please have a look at the latest changes. 

Perhaps we have to do filter `allowed_block_types` in PHP instead of doing it client-side?